aptos-move/framework/supra-framework/sources/automation_registry.move

```diff
@@ -0,0 +1,152 @@
+    /// Default upper gas duration in seconds
+    const DEFAULT_DURATION_UPPER_LIMIT: u64 = 2592000;
+    /// Registry resource creation seed
+    const REGISTRY_RESOURCE_SEED: vector<u8> = b"supra_framework::automation_registry";
+
+    /// It tracks entries both pending and completed, organized by unique indices.
+    struct AutomationRegistry has key, store {
+        /// The current unique index counter for registered tasks. This value increments as new tasks are added.
+        current_index: u64,
+        /// Automation task gas limit
+        automation_gas_limit: u64,
+        /// Automation task duration upper limit.
+        duration_upper_limit: u64,
+        /// It's resource address which is use to deposit user automation fee
+        registry_fee_address: address,
+        /// Resource account signature capability
+        registry_fee_address_signer_cap: SignerCapability,
+        /// A collection of automation task entries that are active state.
+        tasks: EnumerableMap<u64, AutomationTaskMetaData>,
+    }
+
+    #[event]
+    /// `AutomationTaskMetaData` represents a single automation task item, containing metadata.
+    struct AutomationTaskMetaData has copy, store, drop {
+        /// Automation task index in registry
+        id: u64,
+        /// The address of the task owner.
+        owner: address,
+        /// The function signature associated with the registry entry.
+        payload_tx: vector<u8>,
+        /// Expiry of the task, represented in a timestamp in second.
+        expiry_time: u64,
+        /// The transaction hash of the request transaction.
+        tx_hash: vector<u8>,
+        /// Max gas amount of automation task
+        max_gas_amount: u64,
+        /// Maximum gas price cap for the task
+        gas_price_cap: u64,
+        /// Registration epoch number
+        registration_epoch: u64,
+        /// Registration epoch time
+        registration_time: u64,
+        /// Flag indicating whether the task is active.
+        is_active: bool
+    }
+
+    // todo : this function should call during initialzation, but since we already done genesis in that case who can access the function
+    fun initialize(supra_framework: &signer) {
+        system_addresses::assert_supra_framework(supra_framework);
+
+        let (registry_fee_resource_signer, registry_fee_address_signer_cap) = account::create_resource_account(
+            supra_framework,
+            REGISTRY_RESOURCE_SEED
+        );
+
+        move_to(supra_framework, AutomationRegistry {
+            current_index: 0,
+            automation_gas_limit: DEFAULT_AUTOMATION_GAS_LIMIT,
+            duration_upper_limit: DEFAULT_DURATION_UPPER_LIMIT,
+            registry_fee_address: signer::address_of(&registry_fee_resource_signer),
+            registry_fee_address_signer_cap,
+            tasks: enumerable_map::new_map(),
+        })
+    }
+
+    // todo withdraw amount from resource accoun to specified address
```

Comment:
I assume this if to refund for canceled tasks.

Reply:
Yeah, correct.